specification.md

```diff
@@ -694,6 +695,53 @@ to `null`.
 
 ### Job
 
+#### State Model
+
+Job instances are, in this API, stateful objects.  A job's state can be
+inspected via the `job.getStatus()` method which will return a `JobStatus`
+instance on which the job's state is available as an attribute.  State
+transitions can also be received via callbacks
+
+TODO: how are callbacks registered?
+
+An implementation MUST ensure that job state transitions occur according to the
+following state model:  a job is created in an initial state `NEW`.  When the
+job is accepted by the backend for execution, it will enter the state `QUEUED`.
+When the job is being executed and consumes resources, it enters the `ACTIVE`
+state.  Upon completion, it will enter the `COMPLETED` state which is a final
+state.
+
+At any point in time (until the job is final), the job can enter the `FAILED`
+state on error conditions.  That state is also reached when the job completes
+execution with an error code, but can also indicate a backend error, or
+a library error of any kind.  The `FAILED` state is final.
+
+At any point in time (until the job is final), the job can enter the `CANCELED`
+state as reaction to the `job.cancel()` call.  Note that the transition to
+`CANCELED` is not immediate when calling that method, but the state transition
+only occurs once the backend is enacting that request.  
+
+The `ACTIVE` state is the only state where the job will consume resources.
+
```

Comment:
I don't see any definition for PENDING or RESUMED but  they are in the state model diagram.  Not sure which one is right (the diagram or the text).

Reply:
Yeah, that suspend/resume is inconsistent, will fix.

Reply:
suspend/resume are gone.